src/main/java/com/minecolonies/coremod/quests/rewards/HappinessReward.java

```diff
@@ -0,0 +1,70 @@
+package com.minecolonies.coremod.quests.rewards;
+
+import com.google.gson.JsonObject;
+import com.minecolonies.api.colony.IColony;
+import com.minecolonies.api.quests.IColonyQuest;
+import com.minecolonies.api.quests.IQuestReward;
+import com.minecolonies.coremod.colony.CitizenData;
+import net.minecraft.world.entity.player.Player;
+
+import static com.minecolonies.api.util.constant.HappinessConstants.QUEST;
+import static com.minecolonies.api.util.constant.QuestParseConstant.*;
+
+/**
+ * Happiness inducing reward.
+ */
+public class HappinessReward implements IQuestReward
+{
+    /**
+     * Happiness boost.
+     */
+    private final int qty;
+
+    /**
+     * Days to hold.
+     */
+    private final int days;
+
+    /**
+     * Target to affect.
+     */
+    private final int target;
+
+    /**
+     * Setup the research reward.
+     */
+    public HappinessReward(final int target, final int qty, final int days)
```

Comment:
Would it possibly be a good idea to expose creating these somehow in the API?

Reply:
Happiness changes?